awx/main/models/credential/__init__.py

```diff
@@ -504,6 +509,15 @@ class Meta:
                     'Ansible Tower documentation for example syntax.')
     )
 
+    @classmethod
+    def from_db(cls, db, field_names, values):
+        instance = super(CredentialType, cls).from_db(db, field_names, values)
+        if instance.managed_by_tower and instance.native_source:
```

Comment:
Why would a credential type be `managed_by_tower` but yet not have `native_source` set?

Reply:
It wouldn't :)

Reply:
Would it make more sense to call this `ManagedCredentialType`?